Review Request

dsipetcher opened the pull request.

dsipetcher committed:
modified:   apps/client/src/features/editor/components/bubble-menu/bubble-menu.tsx
	modified:   apps/client/src/features/editor/page-editor.tsx
	modified:   apps/client/src/lib/api-client.ts
	modified:   apps/server/src/core/page/page.controller.ts
	modified:   apps/server/src/core/page/page.module.ts
	modified:   apps/server/src/database/database.module.ts
	modified:   apps/server/src/database/types/db.d.ts
	modified:   apps/client/src/features/editor/components/bubble-menu/bubble-menu.tsx
	modified:   apps/client/src/features/editor/page-editor.tsx
	modified:   apps/server/src/app.module.ts
	modified:   apps/server/src/core/casl/interfaces/page-ability.type.ts
	modified:   apps/server/src/core/page/page.controller.ts
	modified:   apps/server/src/core/page/page.module.ts
	modified:   apps/server/src/database/database.module.ts
	modified:   apps/server/src/database/types/db.d.ts
	modified:   apps/server/src/integrations/export/export.service.ts
	modified:   package.json
	modified:   pnpm-lock.yaml
	apps/client/src/features/editor/components/bubble-menu/ContextMenu.tsx
	apps/client/src/features/editor/components/bubble-menu/SearchMenu.tsx
	apps/client/src/features/editor/extensions/custom-paragraph.ts
	apps/client/src/features/editor/utils/getSelectedBlockId.ts
	apps/server/src/core/casl/abilities/block-ability.factory.ts
	apps/server/src/core/casl/interfaces/block-ability.type.ts
	apps/server/src/core/page/dto/save-block-permission.dto.ts
	apps/server/src/core/page/services/block-permission.service.ts
	apps/server/src/database/repos/page/BlockMember.repo.ts
	apps/server/src/database/services/db.service.ts

dsipetcher committed:
new file:   apps/client/src/features/editor/components/bubble-menu/ContextMenu.tsx
	new file:   apps/client/src/features/editor/components/bubble-menu/SearchMenu.tsx
	modified:   apps/client/src/features/editor/components/bubble-menu/bubble-menu.tsx
	new file:   apps/client/src/features/editor/extensions/custom-paragraph.ts
	modified:   apps/client/src/features/editor/page-editor.tsx
	new file:   apps/client/src/features/editor/utils/getSelectedBlockId.ts
	modified:   apps/server/src/app.module.ts
	new file:   apps/server/src/core/casl/abilities/block-ability.factory.ts
	new file:   apps/server/src/core/casl/interfaces/block-ability.type.ts
	modified:   apps/server/src/core/casl/interfaces/page-ability.type.ts
	new file:   apps/server/src/core/page/dto/save-block-permission.dto.ts
	modified:   apps/server/src/core/page/page.controller.ts
	modified:   apps/server/src/core/page/page.module.ts
	new file:   apps/server/src/core/page/services/block-permission.service.ts
	modified:   apps/server/src/database/database.module.ts
	new file:   apps/server/src/database/repos/page/BlockMember.repo.ts
	new file:   apps/server/src/database/services/db.service.ts
	modified:   apps/server/src/database/types/db.d.ts
	modified:   apps/server/src/integrations/export/export.service.ts
	modified:   package.json
	modified:   pnpm-lock.yaml

artyom-kalman commented on docker-compose.yml:
Все переменные окружения лучше перенести в .env файл

artyom-kalman commented on apps/server/src/core/page/services/block-permission.service.ts:
```diff
@@ -0,0 +1,27 @@
+
+@Injectable()
+export class BlockPermissionService {
+  constructor(private readonly db: DbService) {}
```
Тут можно использовать @InjectKysely() private readonly db: KyselyDB вместо DbService

artyom-kalman commented on apps/server/src/database/repos/page/BlockMember.repo.ts:
Иван говорил, что комментарии лучше на английском писать. Я в целом пытаюсь их избегать по возможности. Так что предлагаю либо удалить, либо переписать на английском

artyom-kalman commented on apps/server/src/database/services/db.service.ts:
Предполагаю, что этот сервис нам не нужен. Вся работа с БД у нас через @InjectKysely() private readonly db: KyselyDB.

artyom-kalman commented on apps/server/src/core/casl/abilities/block-ability.factory.ts:
```diff
@@ -0,0 +1,47 @@
+    private readonly blockPermissionRepo: BlockPermissionRepo,
+  ) {}
+
+  /**
+   * Получить права пользователя для блока
+   */
```
Комментарии

artyom-kalman commented on apps/server/src/app.module.ts:
```diff
@@ -16,6 +16,7 @@ import { ExportModule } from './integrations/export/export.module';
 import { ImportModule } from './integrations/import/import.module';
 import { SecurityModule } from './integrations/security/security.module';
 import { TelemetryModule } from './integrations/telemetry/telemetry.module';
+import { BlockPermissionRepo } from '../../../apps/server/src/database/repos/page/BlockMember.repo';
```
Этот испорт используется? Если нет, надо убрать

artyom-kalman commented on apps/client/src/features/editor/page-editor.tsx:
```diff
@@ -291,7 +321,24 @@ export default function PageEditor({
+          zIndex: 9999,
+        }}
+        onClick={(e) => e.stopPropagation()}
+        onContextMenu={(e) => e.preventDefault()} // предотвращаем повторный вызов
```
Комментарии

artyom-kalman commented on apps/client/src/features/editor/page-editor.tsx:
```diff
@@ -190,12 +211,17 @@ export default function PageEditor({
+            // @ts-ignore
+            setEditor(editor);
+            editor.storage.pageId = pageId;
+            setPageId(pageId); // <<<<< Вот здесь сохраняем!
```
Комментарии

artyom-kalman commented on apps/client/src/features/editor/components/bubble-menu/ContextMenu.tsx:
Предлагаю по всему проекту придерживаться одному стилю назвиния файлов. На фронте сейчас все файлы названы строными буквами через тире: foo-baz.ts

artyom-kalman commented on apps/client/src/features/editor/components/bubble-menu/SearchMenu.tsx:
```diff
@@ -0,0 +1,153 @@
+    };
+
+    try {
+      const response = await axios.post(`${API_URL}/api/pages/blockPermissions`, payload);
```
Все API вызовы лучше прописать в отдельном сервисе, чтобы всё было в одном файле. Для премера можешь посмотреть page-service.ts и page-query.ts

artyom-kalman commented on apps/client/src/features/editor/components/bubble-menu/SearchMenu.tsx:
```diff
@@ -0,0 +1,153 @@
+          </ScrollArea.Autosize>
+        ) : (
+          <Text size="sm" c="dimmed" ta="center">
+            Пользователи не найдены
```
Надписи лучше тоже писать на англйиском, а ещё лучше использовать useTranslation: тогда можно будет в будущем добавлсять переводы для любой надписи. Пример есть space-sidebar.tsx и ещё много где по проекту

dsipetcher committed:
modified:   apps/client/src/features/editor/page-editor.tsx

dsipetcher committed:
modified:   apps/client/src/features/editor/components/bubble-menu/bubble-menu.tsx
	renamed:    apps/client/src/features/editor/components/bubble-menu/ContextMenu.tsx -> apps/client/src/features/editor/components/bubble-menu/context-menu.tsx
	renamed:    apps/client/src/features/editor/components/bubble-menu/SearchMenu.tsx -> apps/client/src/features/editor/components/bubble-menu/search-menu.tsx
	renamed:    apps/client/src/features/editor/utils/getSelectedBlockId.ts -> apps/client/src/features/editor/utils/get-selected-block-id.ts

artyom-kalman committed:
Add migartion for blocks

dsipetcher committed:
modified:   apps/client/src/features/editor/full-editor.tsx
	modified:   apps/client/src/features/editor/page-editor.tsx
	modified:   apps/client/src/lib/api-client.ts
	modified:   apps/client/src/pages/page/page.tsx
	modified:   apps/client/tsconfig.json
	modified:   apps/server/src/app.module.ts
	modified:   apps/server/src/core/casl/abilities/block-ability.factory.ts
	new file:   apps/server/src/core/page/dto/update-page-block.dto.ts
	modified:   apps/server/src/core/page/dto/update-page.dto.ts
	new file:   apps/server/src/core/page/extract-page-blocks.ts
	modified:   apps/server/src/core/page/page.controller.ts
	modified:   apps/server/src/core/page/page.module.ts
	new file:   apps/server/src/core/page/services/page-blocks.service.ts
	modified:   apps/server/src/core/page/services/page.service.ts
	modified:   apps/server/src/database/database.module.ts
	new file:   apps/server/src/database/kysely.provider.ts
	renamed:    apps/server/src/database/repos/page/BlockMember.repo.ts -> apps/server/src/database/repos/page/block-member.repo.ts
	modified:   apps/server/src/database/types/db.d.ts
	modified:   apps/server/src/database/types/kysely.types.ts
	new file:   apps/server/src/database/types/page-block.types.ts

artyom-kalman committed:
Add migration for block persmissions

artyom-kalman committed:
Update migrations for blocks

artyom-kalman committed:
Fix block permissions endpoint bag

dsipetcher committed:
Merge pull request #13 from creobitcom/feat/blocks

Migrations fix

dsipetcher committed:
modified:   apps/server/src/core/page/page.controller.ts
	modified:   apps/server/src/core/page/services/block-permission.service.ts
	modified:   apps/server/src/database/types/db.d.ts

dsipetcher committed:
Merge branch 'dev.andrew' of https://github.com/creobitcom/docmost into dev.andrew

dsipetcher committed:
modified:   apps/client/src/features/editor/extensions/custom-paragraph.ts
	new file:   apps/client/src/features/editor/extensions/unique-id.ts
	modified:   apps/client/src/features/editor/page-editor.tsx
	modified:   apps/client/src/lib/api-client.ts
	modified:   apps/server/src/core/page/dto/update-page-block.dto.ts
	modified:   apps/server/src/core/page/extract-page-blocks.ts
	modified:   apps/server/src/core/page/page.controller.ts
	modified:   apps/server/src/core/page/services/page-blocks.service.ts
	modified:   apps/server/src/core/page/services/page.service.ts
	modified:   apps/server/src/database/migrations/20250505T124913-blocks.ts
	modified:   apps/server/src/database/types/db.d.ts
	modified:   package.json
	modified:   pnpm-lock.yaml

dsipetcher committed:
modified:   apps/server/src/core/page/services/page.service.ts

dsipetcher committed:
deleted:    apps/client/src/features/editor/extensions/unique-id.ts

dsipetcher committed:
modified:   apps/client/src/features/editor/extensions/custom-paragraph.ts
	modified:   apps/client/src/features/editor/page-editor.tsx
	modified:   apps/server/src/core/page/services/page.service.ts

dsipetcher committed:
modified:   apps/client/src/features/editor/components/bubble-menu/search-menu.tsx
	modified:   apps/client/src/lib/api-client.ts

dsipetcher committed:
modified:   apps/client/src/features/editor/components/bubble-menu/bubble-menu.tsx
	modified:   apps/client/src/features/editor/components/bubble-menu/search-menu.tsx
	modified:   apps/client/src/features/editor/page-editor.tsx
	modified:   apps/client/src/features/page/services/page-service.ts
	modified:   apps/client/src/lib/api-client.ts
	modified:   apps/server/src/core/page/page.controller.ts
	modified:   apps/server/src/core/page/services/page-blocks.service.ts
	modified:   apps/server/src/core/page/services/page.service.ts
	modified:   apps/server/src/database/database.module.ts
	modified:   apps/server/src/database/types/db.d.ts
